File: src/gestorAplicacion/personas/Autor.java
package gestorAplicacion.personas;
import java.time.LocalDate;
import java.util.ArrayList;
import gestorAplicacion.obras.Libro;

public class Autor extends Persona {
	//ATRIBUTOS DE CLASE
	private static final long serialVersionUID = 1L;
	static private ArrayList<Autor>autores;
	
	//ATRIBUTOS INSTANCIA
	private ArrayList<Libro>libros;
	
	//CONSTRUCTORES
	public Autor() {};
	public Autor (int id, String nombre, LocalDate nacimiento, String paisOrigen){
		super(id,nombre,nacimiento,paisOrigen);
		autores.add(this);
		
	}
	public Autor (int id,String nombre,LocalDate nacimiento, String paisOrigen,boolean vivo){ //Constructor si el Autor no esta vivo
		super(id,nombre, nacimiento, paisOrigen);
		super.vivo=vivo;
		autores.add(this);
	}
	
	
	
	//GETTERS SETTERS
	public static ArrayList<Autor> getAutores() {
		return autores;
	}
	public static void setAutores(ArrayList<Autor> autores) {
		Autor.autores = autores;
	}
	public ArrayList<Libro> getLibros() {
		return libros;
	}
	public void setLibros(ArrayList<Libro> libros) {
		this.libros = libros;
	}
	
	
}

File: Práctica 2/src/gestorAplicacion2/momentaneo/Autor.py


from Persona import Persona


class Autor(Persona):
    #Atributo de clase
    _autores=[]

    #Constructor
    def __init__(self,nombre='Anonimo',id=0,nacimiento='...',pais='...',vivo=True,libros=None):
        if libros is None:
            self.libros=[]
        else:
            self.libros=libros
        super().__init__(nombre,id,"na",'na','na',nacimiento,pais,vivo)
        '''a=True
        for x in Autor._autores:
            if x.getId() == self.getId():
                a= False

                break
        if a == True:
            Autor._autores.append(self)
        else:
            print("El autor ya se encuentra registrado")'''


    #metodos
    def infoPersonal(self):
        c=""
        if self.nombre == 'Anonimo':
            c= "El autor es anonimo"
        elif self.isVivo() == True :
            c = f'INFORMACION DEL AUTOR: \n Nombre: {self.nombre} \n Pais de Origen: {self.pais} \n Fecha de nacimiento: {self.nacimiento} \n ¿Vivo?: SI'
        elif self.isVivo() == False:
            c = f'INFORMACION DEL AUTOR: \n Nombre: {self.nombre} \n Pais de Origen: {self.pais} \n Fecha de nacimiento: {self.nacimiento} \n ¿Vivo?: NO'
        return c
    
    

    
    @classmethod
    def getAutores(cls):
        return Autor._autores
    
    def getLibros(self):
        return self.libros
    def getLibros(self,l):
        self.libros = l